newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -268,17 +270,52 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     }
   }
 
-  _persistValuesToLocalStorage(preferences: Preferences) {
+  _persistValuesToLocalStorage(values: PreferencesValues) {
     try {
-      localStorage.setItem(
-        LocalStorageItem,
-        JSON.stringify(preferences.values)
-      );
+      localStorage.setItem(LocalStorageItem, JSON.stringify(values));
     } catch (e) {
       console.warn('Unable to persist preferences', e);
     }
+  }
+
+  _getLastUsedPath(project: gdProject, kind: ResourceKind) {
+    try {
```

Comment:
We don't expect any exception, you can remove the try catch.
Add explicitly the return value in all cases, including when the path is not found.